### PR TITLE
8301143: [TESTBUG] jfr/event/sampling/TestNative was backported to JDK8u without proper native wrapper

### PR DESCRIPTION
**Pull request by judovana**
This test was brought in by 8223147: JFR Backport and never passed. Despite not being wrong nor having jfr broken in jdk8. jdk8 testsuite simply lacks native support cases.

This is adding the shell wrapper which is compiling and properly setting up the native longSleep  in native shared object.

This commit was tested only on linux.
After this commit whole jdk_jfr group passes

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8301143](https://bugs.openjdk.org/browse/JDK-8301143): [TESTBUG] jfr/event/sampling/TestNative was backported to JDK8u without proper native wrapper

### Reviewers
 * [Severin Gehwolf](https://openjdk.org/census#sgehwolf) (@jerboaa - **Reviewer**)
 * [Paul Hohensee](https://openjdk.org/census#phh) (@phohensee - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk8u-dev pull/235/head:pull/235` \
`$ git checkout pull/235`

Update a local copy of the PR: \
`$ git checkout pull/235` \
`$ git pull https://git.openjdk.org/jdk8u-dev pull/235/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 235`

View PR using the GUI difftool: \
`$ git pr show -t 235`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk8u-dev/pull/235.diff">https://git.openjdk.org/jdk8u-dev/pull/235.diff</a>

</details>

### CODE REVIEW

**Comment by judovana** [comment]
Would be nice if original authors @egahlin and @mgronlun doublechekced if the test is still testing  the same

**Comment by judovana** [comment]
This do not have yet bugid. Will create and adapt as needed if this fix will prove worthy.

**Comment by zzambers** [comment]
Yy, shell wrapper will be necessary to make this work on JDK8.  (This wrapper is inspired by [1], I think.)
I have 2 remarks:

MacOS testing
-  do we want to enable this test on MacOS? (if so, it should probably be tested there)

Linux multilib
- on linux there is also multilib scenario, which is a bit tricky
- basically when building native test libs for 32-bit JDK running on 64-bit system, compiler flag is needed to build 32-bit test libs
- for hotspot tests this case is now detected in test_env.sh file (See this PR for details [2])
- question is: should this be done in test or should there be something like test_env.sh for jdk tests as well?

[1] https://github.com/openjdk/jdk8u-dev/blob/a5bd9018d5671f2399272b680e83239154443b11/jdk/test/jdk/tools/launcher/JliLaunchTest.sh
[2] https://github.com/openjdk/jdk8u-dev/pull/173

**Comment by judovana** [comment]
Thaxn for the ==/=. Fixed now.

The multilib chaneg sounds jsut like 
```
source ${JDK_TOPDIR}/hotspot/test/test_env.sh
...
$gcc ...${CFLAGBITS} ...
```
Which sounds like safe tor. Thanx a lot!

As for MacOs, it should work. I can exclude the Darwin from enumerating. I do not have any mAcOs around.

**Comment by judovana** [comment]
PAss is now visible at: https://github.com/judovana/jdk-tester/actions/runs/3998180323/jobs/6879504748#step:12:406

**Comment by judovana** [comment]
@egahlin ,  @mgronlun  -  although not an jdk8 reviewrs, may you eyball please?
@phohensee , may I be so bold and kindly ask you for official review? (sorry, you popped out inhttps://github.com/openjdk/jdk8u-dev/pull/234)

File: jdk/test/jdk/jfr/event/sampling/TestNative.sh [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+if [ "x$TESTJAVA" = "x" ]; then
+  TESTJAVA=$(dirname $(dirname $(readlink -f $(which java))))
+fi
+
+JDK_TOPDIR="${TESTROOT}/.."
+JAVAC="${TESTJAVA}/bin/javac"
+JAVA="${TESTJAVA}/bin/java"
+TEST_ENV_SH="${JDK_TOPDIR}/../hotspot/test/test_env.sh"
+
+ls -l "${TEST_ENV_SH}"
```
Comment (jerboaa) [inline jdk/test/jdk/jfr/event/sampling/TestNative.sh:34]: This looks like a debug aid. Please remove before pushing.

**Review by jerboaa — APPROVED** [review]
Looks OK to me.

File: jdk/test/jdk/jfr/event/sampling/TestNative.sh (hunk 1 above)
Comment (judovana) [reply jdk/test/jdk/jfr/event/sampling/TestNative.sh:34]: Nope. It is here to fail the test if the file is missing.
See below, the +e/ source -e - the test_env.sh is not designed to run with -e. 

Ok to go  with this in?

Comment (jerboaa) [reply jdk/test/jdk/jfr/event/sampling/TestNative.sh:34]: Fair enough. Feel free to proceed.

**Review by phohensee — APPROVED** [review]
Lgtm.

**Pull request closed without merging**